mozjs/src/gc/root.rs

```diff
@@ -43,7 +43,17 @@ impl<'a, T: 'a + RootKind> RootedGuard<'a, T> {
     }
 
     pub fn handle_mut(&mut self) -> MutableHandle<T> {
-        unsafe { MutableHandle::from_marked_location(self.deref_mut()) }
+        unsafe { MutableHandle::from_marked_location(self.as_ptr()) }
+    }
+
+    pub fn as_ptr(&self) -> *mut T {
+        // SAFETY: self.root points to an inbounds allocation
+        unsafe { (&raw mut (*self.root).ptr).cast() }
+    }
+
+    /// Safety: GC must not run during the lifetime of the returned reference.
+    pub unsafe fn as_mut<'b: 'a>(&mut self) -> &'b mut T {
+        &mut *(self.as_ptr())
```

Comment:
Those lifetimes are wrong as per https://github.com/servo/mozjs/pull/573#discussion_r2013405578

Reply:
Oops, forgot I copy and pasted this, thanks for catching this twice now

Reply:
Updated again as per discussion in https://github.com/servo/mozjs/issues/569